snarkos/ledger/server.rs

```diff
@@ -184,17 +185,26 @@ impl<N: Network> Server<N> {
             .and(warp::path!("testnet3" / "transaction" / ..))
             .and(warp::path::param::<N::TransactionID>())
             .and(warp::path::end())
-            .and(with(ledger))
+            .and(with(ledger.clone()))
             .and_then(Self::get_transaction);
 
         // POST /testnet3/transaction/broadcast
         let transaction_broadcast = warp::post()
             .and(warp::path!("testnet3" / "transaction" / "broadcast"))
             .and(warp::body::content_length_limit(10 * 1024 * 1024))
             .and(warp::body::json())
-            .and(with(ledger_sender))
+            .and(with(ledger_sender.clone()))
             .and_then(Self::transaction_broadcast);
 
+        // POST /testnet3/program/deploy
```

Comment:
the comment doesn't match with the url, you should change one of them (`/testnet3/program/deploy` does sound better, but probably requires changes in current clients)
```suggestion
        // POST /testnet3/deploy
```